src/frequenz/client/reporting/_client.py

```diff
@@ -252,3 +253,33 @@ def dt2ts(dt: datetime) -> PBTimestamp:
         except grpcaio.AioRpcError as e:
             print(f"RPC failed: {e}")
             return
+
+    @staticmethod
+    def calculate_energy_metric(
+        metric_samples: list[MetricSample], time_interval: timedelta
+    ) -> pd.Series:
+        """
+        Calculate the energy metric with AC_ACTIVE_POWER values and time interval.
+
+        Args:
+            metric_samples: A list of MetricSample objects, potentially containing various metrics.
+            time_interval: A timedelta object representing the time interval for energy calculation.
+
+        Returns:
+        A pandas Series with timestamps as index and the calculated energy in Wh as values.
+        """
+        ac_active_power_samples = [
+            sample for sample in metric_samples if sample.metric == "AC_ACTIVE_POWER"
+        ]
+
+        ac_active_power_series = pd.Series(
+            data=[sample.value for sample in ac_active_power_samples],
+            index=[sample.timestamp for sample in ac_active_power_samples],
+        )
+
+        # Convert the timedelta to hours
+        hours = time_interval.total_seconds() / 3600.0
+
+        energy_wh_series = ac_active_power_series * hours
```

Comment:
I don't think this is what is wanted. Usually the energy metric is integrated over time, so the power would need to be converted to energy and then cumulatively added.